### PR TITLE
Find root of nary tree the fastest

### PR DESCRIPTION
**Pull request by altayhunoglu**
https://github.com/spiralgo/algorithms/issues/257

image

### CODE REVIEW

**Review by ErdemT09 — COMMENTED** [review]
How is this a linear time algorithm as wanted in the question?

File: src/main/java/algorithms/curated170/easy/largestuniquenumber/LargestUniqueNumberFaster.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package algorithms.curated170.easy.largestuniquenumber;
+
+ 
+public class LargestUniqueNumberFaster {
```
[the remark this answers is not in the document]
Comment (ErdemT09) [reply src/main/java/algorithms/curated170/easy/largestuniquenumber/LargestUniqueNumberFaster.java:4]: No, I mean the root of n-ary tree algorithm as in the PR.

[the remark this answers is not in the document]
Comment (ErdemT09) [reply src/main/java/algorithms/curated170/easy/largestuniquenumber/LargestUniqueNumberFaster.java:4]: But I kind of get it now. The first loop can have operations at maximum I think.

[the remark this answers is not in the document]
Comment (altayhunoglu) [reply src/main/java/algorithms/curated170/easy/largestuniquenumber/LargestUniqueNumberFaster.java:4]: The older codes may or may not consider a right examination of time and space efficiency and follow-ups and they are rather focused on correctness.
  It is one of the reasons we should add the most efficient one.

**Comment by ErdemT09** [comment]
Instead of this:
```
        Node root = null;
        // the value of the root node is `valueSum`
        for (Node node : tree) {
            if (node.val == valueSum) {
                root = node;
                break;
            }
        }
        return root;
```
We can simplify to not creating a new object.

```
        for (Node node : tree) {
            if (node.val == valueSum) {
                return node;
            }
        }
        return null;
```

**Review by ErdemT09 — APPROVED** [review]
Approved.

**Pull request merged**